### PR TITLE
Remove hack when creating and deleting reports

### PR DESCRIPTION
**Pull request by ayolab**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by TheMaskedTurtle — APPROVED** [review]
I think we should wait for the release with springboot to not block it but then ok !

File: src/test/java/com/powsybl/ws/commons/computation/service/ReportServiceTest.java
Comment (TheMaskedTurtle) [inline src/test/java/com/powsybl/ws/commons/computation/service/ReportServiceTest.java:1]: URL assertions should be changed tests don't pass

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: src/test/java/com/powsybl/ws/commons/computation/service/ReportServiceTest.java
Comment (ayolab) [reply src/test/java/com/powsybl/ws/commons/computation/service/ReportServiceTest.java:1]: Done

**Review by dbraquart — COMMENTED** [review]
code: OK
tests:on going

Note: When I upgrade voltage-init-server to use this PR, I have 2 compilation errors ! I had to fix them to run the server.  And I also had a test failure (testStop).

File: src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java [hunk 1]
```diff
@@ -203,13 +203,19 @@ protected R run(C runContext, UUID resultUuid, AtomicReference<ReportNode> rootR
...
                     .withUntypedValue("providerToUse", Objects.requireNonNullElse(provider, "")).add();
             // Delete any previous computation logs
             observer.observe("report.delete",
-                    runContext, () -> reportService.deleteReport(runContext.getReportInfos().reportUuid(), reportType));
+                    runContext, () -> {
+                        try {
+                            reportService.deleteReport(runContext.getReportInfos().reportUuid());
+                        } catch (Exception e) {
+                            LOGGER.error("Failed to delete report", e);
+                        }
```
Comment (TheMaskedTurtle) [inline src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:217]: This changes the behavior, are we sure this is what we want ?

Comment (TheMaskedTurtle) [reply src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:217]: Here we catch all errors even 5xx not only 404

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**